### DISCUSSION TITLE
Announcing Keyark MRF Search web application -- locate the MRFs you need -- free to use

### DISCUSSION THREAD

**Discussion by dgolden** [CMSgov/price-transparency-guide#670] [Show and tell]
Have you been stymied trying to figure out _which_ In-Network MRF, out of thousands published, has the data you need? Keyark has a solution for you.

**Announcing free use of the Keyark [MRF Search](https://www.healthplanmrf.com/mrfsearch/mrfsearchapp) web application during our beta test period. 
With just your web browser, you can find and download MRF data.**

Keyark’s web crawler creates a directory of In-Network MRFs for leading payers, which can be explored through our MRF Search application. Currently the following two payers are available – UnitedHealthcare and Cigna – with more to come. 

The screenshot below shows a search for all UnitedHealthcare MRFs that fit the following criteria:
-	Contains rates for CP 45378, a fiber optic colonoscopy
-	Individual providers in Nashville, TN, whose practice is located in the set of Zip codes that begin with 3720 (central Nashville)
-	Specialty in gastroenterology
 
If you know the EIN, you can further narrow the search results.

<img width="918" alt="mrf_search_taxononomy(gastroenterology)_ZipCode(372)_BillingCode(45378)" src="https://user-images.githubusercontent.com/6385204/232385331-d75273cc-8789-42a6-a4c8-374dd761cb3f.png">

The result is a text file with the full URL for all MRFs that have relevant data. Copy a URL into our companion product [MRF Explorer](https://www.healthplanmrf.com/mrfexplorer/mrfreportapp) to now fetch rate information:

<img width="918" alt="mrf_explorer_npi_selection_zip_372_wildcard" src="https://user-images.githubusercontent.com/6385204/232387246-a59c62b8-1599-4162-92c6-53242e2d5883.png">
MRF Explorer, which is also free to use during our beta period, saves the MRF data to a CSV file, as shown here in Excel:

<img width="648" alt="mrf_explorer_npi_selection_results_zip_372_wildcard" src="https://user-images.githubusercontent.com/6385204/232389047-a9826623-afc0-49dc-9175-be7278bf41ab.png">

In just 2-3 minutes you can execute these steps to gain transparency into the rate data that most interests you.

For more information on MRF Explorer please visit https://www.healthplanmrf.com/

**Comment by brendanyounger** [comment]
@dgolden I tried out the search tool with NPI 1003895095 and CPT 99213. It appears to be timing out. At least, I gave up and closed the tab after a few minutes.

**Comment by dgolden** [comment]
GIve it more time. Some of the MRFs are enormous, and the payer may also
throttle.

You can also email the URL to me --  dgolden -at- keyark dot com -- and
I'll check out the experience.
